mlflow/promptflow/__init__.py

```diff
@@ -0,0 +1,393 @@
+            ColSpec(type=_parse_pf_type(typ), name=k) for k, typ in model.inputs.items()
+        ]
+        input_schema = Schema(input_columns)
+
+        output_columns = [
+            ColSpec(type=_parse_pf_type(typ), name=k) for k, typ in model.outputs.items()
+        ]
+        output_schema = Schema(output_columns)
+
+        signature = ModelSignature(input_schema, output_schema)
+
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, path)
+    if metadata is not None:
+        mlflow_model.metadata = metadata
+
+    # update flavor info to mlflow_model
+    mlflow_model.add_flavor(
+        FLAVOR_NAME,
+        code=code_dir_subpath,
+        promptflow_version=promptflow.__version__,
+        entry=_MODEL_FLOW_DIRECTORY,
+        **flow_env
+    )
+
+    # append loader_module, data and env data to mlflow_model
+    pyfunc.add_to_model(
+        mlflow_model,
+        loader_module="mlflow.promptflow",
+        conda_env=_CONDA_ENV_FILE_NAME,
+        python_env=_PYTHON_ENV_FILE_NAME,
+        code=code_dir_subpath,
+    )
+
+    # save mlflow_model to path/MLmodel
+    mlflow_model.save(os.path.join(path, MLMODEL_FILE_NAME))
+
+    if conda_env is None:
+        if pip_requirements is None:
+            default_reqs = get_default_pip_requirements()
+            # To ensure `_load_pyfunc` can successfully load the model during the dependency
+            # inference, `mlflow_model.save` must be called beforehand to save an MLmodel file.
+            inferred_reqs = mlflow.models.infer_pip_requirements(
+                path, FLAVOR_NAME, fallback=default_reqs
+            )
+            default_reqs = sorted(set(inferred_reqs).union(default_reqs))
+        else:
+            default_reqs = None
+        conda_env, pip_requirements, pip_constraints = _process_pip_requirements(
+            default_reqs,
+            pip_requirements,
+            extra_pip_requirements,
+        )
+    else:
+        conda_env, pip_requirements, pip_constraints = _process_conda_env(conda_env)
+
+    with open(os.path.join(path, _CONDA_ENV_FILE_NAME), "w") as f:
+        yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
+
+    # Save `constraints.txt` if necessary
+    if pip_constraints:
+        write_to(os.path.join(path, _CONSTRAINTS_FILE_NAME), "\n".join(pip_constraints))
+
+    # Save `requirements.txt`
+    write_to(os.path.join(path, _REQUIREMENTS_FILE_NAME), "\n".join(pip_requirements))
+
+    _PythonEnv.current().to_yaml(os.path.join(path, _PYTHON_ENV_FILE_NAME))
+
+
+def _resolve_env_from_flow(flow_dag_path):
+    with open(flow_dag_path, "r") as f:
+        flow_dict = yaml.safe_load(f)
+    environment = flow_dict.get("environment", {})
+    return environment
+
+
+class _PromptflowModelWrapper:
+    def __init__(self, model):
+        self.model = model
```

Comment:
below you are passing in a path, but here you are treating it like a rehydrated object

Reply:
Thanks, refined.